### DISCUSSION TITLE
How do I call the method of the struct ?

### DISCUSSION THREAD

**Discussion by lz520520** [traefik/yaegi#1531] [Q&A] [answered]
### The following program `sample.go` triggers an unexpected result

```go
package script

import (
	"fmt"
	"github.com/traefik/yaegi/interp"
	"github.com/traefik/yaegi/stdlib"
	"testing"
)


var (
	src2 = `
package plugins_test

import "fmt"



type Test struct {

}

func (t *Test) Run()  {
	fmt.Println("run")
}

`

)

func TestTest(t *testing.T) {
	intp := interp.New(interp.Options{})

	intp.Use(stdlib.Symbols)
	_, err := intp.Eval(src2)
	if err != nil {
		panic(err)
	}


	v, _ := intp.Eval("plugins_test.Test")
	fmt.Println(v.MethodByName("Run"))

}
```

### Expected result

```console
Here is the test of calling the structure method, mainly see the following error
```

### Got

```console
=== RUN   TestTest
<invalid reflect.Value>
--- PASS: TestTest (0.00s)
PASS
```

### Yaegi Version

v0.11.2

### Additional Notes

I want to call the method of this structure, but it seems that I can’t get the method

**Comment by Lcys** [comment]
Has the landlord found a solution?

**Comment by ApolloX939** [comment]
Looks like [https://github.com/traefik/yaegi/blob/master/example/getfunc/getfunc_test.go#L26](https://github.com/traefik/yaegi/blob/master/example/getfunc/getfunc_test.go#L26) might be what you want to try out.

**Comment by mvertes** [comment, marked as answer]
Because of some limitations of Go reflect packages, the resolution of methods can not be performed outside of the interpreter context. What works is the following:

```go
v, err := intp.Eval("plugins_test.Test.Run") // Run method is obtained from within the interpreter, not reflect.
v.Call(nil)
```
